src/mgr/Mgr.cc

```diff
@@ -91,10 +92,16 @@ void MetadataUpdate::finish(int r)
         }
       }
 
+      // skip for monitor who doesn't have host-name yet
+      if (key.first == "mon" && daemon_meta.count("hostname") == 0) {
```

Comment:
i'd suggest move this change into its own commit. and i think we should apply the sanity check to other daemons as well.

Reply:
> i'd suggest move this change into its own commit. and i think we should apply the sanity check to other daemons as well.

Yes, I agree that this check should be applied for every daemon type(mon,osd,mds,mgr). But  I don't quite understand about "move this change into its own commit". Could you explain more plain? Thanks.

Reply:
i mean we should split it into a separated commit.